### PR TITLE
Jami mini swerve

### PR DESCRIPTION
**Pull request by Jdiep7**
Swerve Summer Code integrated into 2023 Robot Code.

Auto folder, commands, swerve module file and drive subsystem for swerve moved into 2023 robot code.
AHRS gyro in drive subsystem also changed to pigeon imu and lamprey encoder changed to CanCoder in swerve module.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.SerialPort.Port;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.config.Config;
+
+public class DriveSubsystem extends SubsystemBase {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/subsystems/DriveSubsystem.java:22]: suggest to change the class name to SwerveDriveSubsystem or SwerveSubsystem.

File: src/main/java/frc/robot/commands/DriveCommand.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.config.Config;
+import frc.robot.subsystems.DriveSubsystem;
+
+public class DriveCommand extends CommandBase {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/DriveCommand.java:13]: Suggest to change the class name to SwerveDriveCommand.

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    // The gyro sensor
+    private final PigeonIMU m_pigeon; 
+
+    // Odometry class for tracking robot pose
+    SwerveDriveOdometry m_odometry;
+
+    /** Get instance of singleton class */
+    public static DriveSubsystem getInstance() {
+        if (instance == null)
+            instance = new DriveSubsystem();
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/subsystems/DriveSubsystem.java:50]: Add subsystemConstructed( ) check here.

File: src/main/java/frc/robot/config/Config.java [hunk 4]
```diff
@@ -90,7 +110,101 @@ public static class CANID {
...
+    public static final double kPYController = 1;
+    public static final double kPThetaController = 1;
+
+    public static final double kMaxTeleopAngularSpeed = Math.PI*2;
+    public static final double FL_ENCODER_OFFSET = -(155 + 180)-3.22;
+    public static final double FR_ENCODER_OFFSET = -(94 + 180);
+    public static final double RL_ENCODER_OFFSET = -(200 + 180)-0.08;
+    public static final double RR_ENCODER_OFFSET = -(135 + 180)-1.24;
+
+    public static final int CANID_FRONT_LEFT_DRIVE = 6;
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/config/Config.java:123]: Those CANIDs could be merged to Config.CANID.

**Review by ErikCald — APPROVED** [review]
This branch is more tested than main. We should pull the changes into main so that the path planner crew can use the swerve subsystem in the path planner commands.

Only 1 small merge conflict to fix before this can be merged.

File: src/main/java/frc/robot/auto/AutoSelector.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Comment (GabbyGenereux) [inline src/main/java/frc/robot/auto/AutoSelector.java:1]: Ideally we wouldn't have non-swerve drive related changes inside this commit. This ensures we're able to keep commits small and manageable, as well as understand what all the changes relate to.

File: src/main/java/frc/robot/commands/FancySwerveGyroReset.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Comment (GabbyGenereux) [inline src/main/java/frc/robot/commands/FancySwerveGyroReset.java:1]: Is this file needed or used anywhere? It doesn't look like it contains anything useful, from my initial glance.

File: src/main/java/frc/robot/robotcontainers/MiniSwerveContainer.java [hunk 7]
```diff
@@ -4,11 +4,22 @@
 
 package frc.robot.robotcontainers;
```
Comment (GabbyGenereux) [inline src/main/java/frc/robot/robotcontainers/MiniSwerveContainer.java:5]: We should rename this file from mini swerve to just swerve container

**Review by GabbyGenereux — APPROVED** [review]
